lib/Controller/VideoController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @copyright Copyright (c) 2021 Frederic Ruget <fred@atlant.is>
+ *
+ * @author Frederic Ruget (douzebis) <fred@atlant.is>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Viewer\Controller;
+
+use Exception;
+
+use OCP\IRequest;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Controller;
+use OCP\L10N\IFactory;
+
+use OCP\Files\File;
+use OCP\Files\IRootFolder;
+use OCP\IUserSession;
+
+class VideoController extends Controller {
```

Comment:
I would prefer an OCS controller.
So it's a real api please :)

See the forms controller example I gave you https://github.com/nextcloud/forms/blob/41853b59f346913116524baff8c806301abdafe9/lib/Controller/ApiController.php#L64

Reply:
@skjnldsv , so `VideoController` should extend `OCSController`, right?
Also, in `routes.php`, should the url include a version number?
E.g.: `'url' => '/video/{apiVersion}/tracks'`

Reply:
Not specifically a version number, but route should be within the ocs  scope

Reply:
https://github.com/nextcloud/server/blob/d9015a8c94bfd71fe484618a06d276701d3bf9ff/apps/accessibility/appinfo/routes.php#L31-L47